src/Processors/Formats/IInputFormat.cpp

```diff
@@ -24,7 +24,6 @@ void IInputFormat::resetParser()
 
 void IInputFormat::setReadBuffer(ReadBuffer & in_)
 {
-    chassert(in); // not supported by random-access formats
```

Comment:
why?

Reply:
Because before `setReadBuffer` now we call `resetReadBuffer` that resets `in` to `nullptr`